### PR TITLE
Pages in states

### PR DESCRIPTION
**Pull request by chrismiceli**
https://www.pgdp.org/~chrismiceli/c.branch/pages_in_states/stats/pages_in_states.php

### CODE REVIEW

**Review by srjfoo — APPROVED** [review]
Looks good to me.

File: stats/pages_in_states.php [hunk 1]
```diff
@@ -59,45 +61,45 @@
...
-$graph->xaxis->SetTickLabels($stage_labels);
-
-$graph->title->SetFont($jpgraph_FF, $jpgraph_FS);
-$graph->yaxis->title->SetFont($jpgraph_FF, $jpgraph_FS);
-$graph->xaxis->title->SetFont($jpgraph_FF, $jpgraph_FS);
-
-// Display the graph
-$graph->Stroke();
+$title = _("Number of pages in various states");
+// $graph->xaxis->title->Set(_("stage"));
```
Comment (cpeel) [inline stats/pages_in_states.php:67]: This needs to be removed too.

**Review by cpeel — COMMENTED** [review]
We should also remove the following line in `INSTALL.md` since GD is no longer required after this:
https://github.com/DistributedProofreaders/dproofreaders/blame/master/SETUP/INSTALL.md#L23

Please also add an entry in the `CHANGELOG.md`.

I'd like for this to be merged in two commits:
1. pages_in_state
2. jpgraph removal

So when you make these updates, please squash your new commits into the appropriate existing one.

**Comment by chrismiceli** [comment]
> 
> 
> We should also remove the following line in `INSTALL.md` since GD is no longer required after this:
> https://github.com/DistributedProofreaders/dproofreaders/blame/master/SETUP/INSTALL.md#L23
> 
> Please also add an entry in the `CHANGELOG.md`.
> 
> I'd like for this to be merged in two commits:
> 
>     1. pages_in_state
> 
>     2. jpgraph removal
> 
> 
> So when you make these updates, please squash your new commits into the appropriate existing one.

My CHANGELOG entries may need some work. I'm not a great technical writer.

File: SETUP/CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,9 @@ see the git history.
 ## R??????
 Scripts supporting this upgrade are in `SETUP/upgrade/16`
 
+* Remove dependency on jpgraph and GD.
+* Introduct new client rendered graphing capabilities that support theming.
+
```
Comment (cpeel) [inline SETUP/CHANGELOG.md:11]: Perhaps:
```
* Removed jpgraph and GD dependencies (chrismiceli)
* Graphs now rendered via client-side Javascript as themed SVGs (chrismiceli)
```

Comment (srjfoo) [reply SETUP/CHANGELOG.md:11]: Won't this go in `SETUP/upgrade/17`, since the plan is to not include it in the 202109 release?

Comment (chrismiceli) [reply SETUP/CHANGELOG.md:11]: Well handle this in the merge to master i believe unless @cpeel wants me to sync this branch up now?

**Review by srjfoo — APPROVED** [review]

File: SETUP/CHANGELOG.md (hunk 2 above)
Comment (cpeel) [reply SETUP/CHANGELOG.md:11]: Correct - I'll take care of this on merge to master.

**Pull request merged**